gldemo/plugin.cpp

```diff
@@ -190,97 +130,80 @@ class gldemo : public threadloop {
-			glUniformMatrix4fv(projectionAttr, 1, GL_FALSE, (GLfloat*)(basicProjection.data()));
+			// Excessive? Maybe.
+			constexpr int LEFT_EYE = 0;
 
-			glBindVertexArray(demo_vao);
+			for(auto eye_idx = 0; eye_idx < 2; eye_idx++) {
 
-			#ifdef USE_ALT_EYE_FORMAT
+				// Offset of eyeball from pose
+				auto eyeball = Eigen::Vector3f((eye_idx == LEFT_EYE ? -ipd/2.0f : ipd/2.0f), 0, 0);
 
-			// Draw things to left eye.
-			glBindTexture(GL_TEXTURE_2D, eyeTextures[0]);
-			glFramebufferTexture(GL_DRAW_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, eyeTextures[0], 0);
-			glBindTexture(GL_TEXTURE_2D, 0);
-			glClearColor(0.6f, 0.8f, 0.9f, 1.0f);
-			glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
-			
-			draw_scene();
+				// Apply head rotation to eyeball offset vector
+				eyeball = head_rotation_matrix * eyeball;
 
-			
-			//glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, idx_vbo);
-			//glDrawElements(GL_TRIANGLES, BLOCKI_NUM_POLYS * 3, GL_UNSIGNED_INT, (void*)0);
-			
-			
-			// Draw things to right eye.
-			glBindTexture(GL_TEXTURE_2D, eyeTextures[1]);
-			glFramebufferTexture(GL_DRAW_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, eyeTextures[1], 0);
-			glBindTexture(GL_TEXTURE_2D, 0);
-			glClearColor(0.6f, 0.8f, 0.9f, 1.0f);
-			glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+				// Apply head position to eyeball
+				eyeball += pose.position;
 
-			draw_scene();
+				// Build our eye matrix from the pose's position + orientation.
+				Eigen::Matrix4f eye_matrix = Eigen::Matrix4f::Identity();
+				eye_matrix.block<3,1>(0,3) = eyeball; // Set position to eyeball's position
+				eye_matrix.block<3,3>(0,0) = pose.orientation.toRotationMatrix();
 
-			#else
-			
-			// Draw things to left eye.
-			glBindTexture(GL_TEXTURE_2D_ARRAY, eyeTextures[buffer_to_use]);
-			glFramebufferTextureLayer(GL_DRAW_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, eyeTextures[buffer_to_use], 0, 0);
-			glBindTexture(GL_TEXTURE_2D_ARRAY, 0);
-			glClearColor(0.6f, 0.8f, 0.9f, 1.0f);
-			glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
-			
-			draw_scene();
+				// Objects' "view matrix" is inverse of eye matrix.
+				auto view_matrix = eye_matrix.inverse();
 
-			
-			//glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, idx_vbo);
-			//glDrawElements(GL_TRIANGLES, BLOCKI_NUM_POLYS * 3, GL_UNSIGNED_INT, (void*)0);
-			
-			
-			// Draw things to right eye.
-			glBindTexture(GL_TEXTURE_2D_ARRAY, eyeTextures[buffer_to_use]);
-			glFramebufferTextureLayer(GL_DRAW_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, eyeTextures[buffer_to_use], 0, 1);
-			glBindTexture(GL_TEXTURE_2D_ARRAY, 0);
-			glClearColor(0.6f, 0.8f, 0.9f, 1.0f);
-			glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+				Eigen::Matrix4f modelViewMatrix = modelMatrix * view_matrix;
+				glUniformMatrix4fv(modelViewAttr, 1, GL_FALSE, (GLfloat*)(modelViewMatrix.data()));
+				glUniformMatrix4fv(projectionAttr, 1, GL_FALSE, (GLfloat*)(basicProjection.data()));
 
-			draw_scene();
+#ifdef USE_ALT_EYE_FORMAT
```

Comment:
Remove because of #127.

Reply:
We will merge #127 before this and I'll rebase/merge up this branch.

Reply:
Done.